src/Aspire.Dashboard/Components/Controls/SelectResourceOptions.razor

```diff
@@ -0,0 +1,55 @@
+﻿@namespace Aspire.Dashboard.Components
+
+@using System.Collections.Concurrent
+@using Aspire.Dashboard.Model
+@using Aspire.Dashboard.Resources
+
+@inject IStringLocalizer<ControlsStrings> ControlsStringsLoc
+
+@typeparam TValue where TValue : notnull
+
+<FluentStack Orientation="Orientation.Vertical">
+    <FluentCheckbox Label="@ControlsStringsLoc[nameof(ControlsStrings.LabelAll)]"
+                    ThreeState="true"
+                    ShowIndeterminate="false"
+                    ThreeStateOrderUncheckToIntermediate="true"
+                    @bind-CheckState:get="@AreAllVisible"
+                    @bind-CheckState:set="@OnAllValuesCheckedChangedInternalAsync"
+    />
+
+    @foreach (var (resourceType, _) in AllValues
+                  .Where(pair => Resources.Any(resource => ResourceContainsValue(resource, pair.Key)))
```

Comment:
This is inefficient but I figured it didn't matter much since this `ResourceContainsValue` isn't that expensive and this is only called on filter open.